client/me/form-base/with-form-base.jsx

```diff
@@ -0,0 +1,126 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import debugFactory from 'debug';
+
+const debug = debugFactory( 'calypso:me:form-base' );
+
+/**
+ * Internal dependencies
+ */
+import notices from 'calypso/notices';
+import user from 'calypso/lib/user';
+
+const withFormBase = ( WrappedComponent ) => {
+	return class extends React.Component {
+		static displayName = `withFormBase(${ WrappedComponent.displayName || WrappedComponent.name })`;
+
+		state = {
+			redirect: false,
+			submittingForm: false,
+			changingUsername: false,
+			usernameAction: 'new',
```

Comment:
I agree. Will

Reply:
Agreed. The components which use the `formBase` mixin rely on state being initialized in the mixin so that's probably the reason why it's there. I'll remove it from here, we just need to remember to add it to the component which uses `usernameAction` when we migrate it to use the HOC as well.

Reply:
fixed via 7153f93